lib/tree.rb

```diff
@@ -1,64 +1,127 @@
 class TreeNode
   attr_reader :key, :value
   attr_accessor :left, :right
-
-   def initialize(key, val)
+  
+  def initialize(key, val)
     @key = key
     @value = val
     @left = nil
     @right = nil
-   end
+  end
 end
 
 class Tree
   attr_reader :root
   def initialize
     @root = nil
   end
-
-  # Time Complexity: 
-  # Space Complexity: 
+  
+  # Time Complexity: O(n)
+  # Space Complexity: O(n)
   def add(key, value)
-    raise NotImplementedError
+    @root = add_helper(@root, key, value)
   end
-
-  # Time Complexity: 
-  # Space Complexity: 
+  
+  def add_helper(current, key, value)
+    return TreeNode.new(key, value) if current.nil?
+    
+    if key < current.key
+      current.left = add_helper(current.left, key, value)
+    else
+      current.right = add_helper(current.right, key, value)
+    end 
+    return current
+  end
+  
+  # Time Complexity: O(n)
+  # Space Complexity: O(log n)
   def find(key)
-    raise NotImplementedError
+    return nil if @root == nil
+    return find_helper(@root, key)
   end
-
-  # Time Complexity: 
-  # Space Complexity: 
+  
+  def find_helper(current, key)
+    return nil if current.nil?
+    return current.value if current.key == key
+    
+    if key < current.key
+      find_helper(current.left, key)
```

Comment:
Just a suggestion for readability

```suggestion
      return find_helper(current.left, key)
```